packages/dara-core/dara/core/interactivity/derived_variable.py

```diff
@@ -211,10 +217,14 @@ def _restore_pydantic_models(func: Callable[..., Any], *args):
         if len(var_arg_idx) == 0:
             for param, arg in zip(parameters, args):
                 typ = param.annotation
-                if typ is not None and typ in encoder_registry:
-                    parsed_args.append(encoder_registry[typ]['deserialize'](arg))
-                elif typ is not None and isclass(typ) and issubclass(typ, BaseModel) and arg is not None:
-                    parsed_args.append(typ(**arg))
+                if not isinstance(arg, BaseTask):
```

Comment:
This section should call `deserialize` from `encoder_registry`, see the `1.1.x` PR for reference

Reply:
thank you for catching those, have now fixed